js/apiClient.js

```diff
@@ -0,0 +1,221 @@
+// Function to download the configuration file
+function downloadConfigFile() {
+    const configData = getConfigurationDataForFile();
+
+    // Create configuration file content string
+    const config = Object.entries(configData)
+        .map(([key, value]) => `${key}=${value}`)
+        .join('\n');
+
+    // Create a blob and trigger download
+    const blob = new Blob([config], { type: 'text/plain' });
+    const url = URL.createObjectURL(blob);
+    const a = document.createElement('a');
+    a.href = url;
+    a.download = 'nzcvm_config.txt'; // Filename for the download
+    document.body.appendChild(a);
+    a.click();
+    document.body.removeChild(a);
+    URL.revokeObjectURL(url);
+}
+
+// Function to trigger the backend model generation and download results
+async function generateModelAndDownload() {
+    // Clear any existing timer interval
+    if (countdownIntervalId) {
+        clearInterval(countdownIntervalId);
+        countdownIntervalId = null;
+    }
+
+    const statusMessage = document.getElementById('status-message');
+    statusMessage.style.color = 'orange'; // Set initial color
+
+    // --- Calculate estimated runtime ---
+    let estimatedSeconds = 0;
+    try {
+        // Read values needed for calculation (ensure they exist and are valid)
+        const extentX = parseFloat(document.getElementById('extent-x').value);
+        const extentY = parseFloat(document.getElementById('extent-y').value);
+        const extentLatlonSpacing = parseFloat(document.getElementById('xy-spacing').value);
+        const extentZmax = parseFloat(document.getElementById('extent-zmax').value);
+        const extentZmin = parseFloat(document.getElementById('extent-zmin').value);
+        const extentZSpacing = parseFloat(document.getElementById('z-spacing').value);
+
+        // Check if calculateGridPoints and calculateApproxRunTime are available
+        if (typeof calculateGridPoints === 'function' && typeof calculateApproxRunTime === 'function') {
+            const gridData = calculateGridPoints(extentX, extentY, extentLatlonSpacing, extentZmax, extentZmin, extentZSpacing);
+            estimatedSeconds = calculateApproxRunTime(gridData.totalGridPoints);
+        } else {
+            console.warn("Calculation functions not found. Skipping timer estimation.");
+            estimatedSeconds = 0; // Default or skip timer
+        }
+    } catch (calcError) {
+        console.error("Error calculating runtime:", calcError);
+        estimatedSeconds = 0; // Default or skip timer on calculation error
+    }
+
+    // --- Check if estimated runtime exceeds the limit ---
+    if (estimatedSeconds >= 600) {
+        statusMessage.innerHTML = 'This website can only generate models that require less than 10 min. Please reduce the number of points in your grid until the run time is less than 10 minutes, or <a href="#" id="download-config-link-for-large-model">download the configuration file</a> and generate the model on your computer.';
```

Comment:
Could we just increase the timeout values in the nginx config to be higher than 10 minutes as well?